File: src-1.x/BYTES.NET/WPF/Converters/NothingToBoolConverter.vb
'import .net namespace(s) required
Imports System.Globalization

Namespace WPF.Converters

    ''' <summary>
    ''' converter returning a boolean based on an object being nothing or not nothing
    ''' </summary>
    ''' <remarks>by default 'true' will be returned if the object is nothing; if parameter is set to 'false', 'true' will be returned if value is not 'Nothing'; if parameter is set to 'true', 'true' will be returned for an "empty string" also</remarks>
    Public Class NothingToBoolConverter

        Implements IValueConverter

        Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert

            If IsNothing(parameter) OrElse String.IsNullOrEmpty(parameter) Then 'no parameter was given

                If IsNothing(value) Then

                    Return True

                Else

                    Return False

                End If

            Else 'a parameter was given

                Try

                    Dim comparing As Boolean = CBool(parameter)

                    If comparing Then

                        If IsNothing(value) = comparing OrElse String.IsNullOrEmpty(value) Then

                            Return True

                        Else

                            Return False

                        End If

                    Else

                        If Not IsNothing(value) Then

                            Return True

                        Else

                            Return False

                        End If

                    End If

                Catch ex As Exception

                    Throw New ArgumentException("Unable to convert value")

                End Try

            End If

        End Function

        Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack

            Return Nothing

        End Function

    End Class

End Namespace

File: src-2023/BYTES.NET.WPF/Converters/NothingToBoolConverter.cs
﻿//import .net namespace(s) required
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Data;

namespace BYTES.NET.WPF.Converters
{
    public class NothingToBoolConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (parameter == null || string.IsNullOrEmpty(parameter.ToString()))
            {
                // No parameter provided, check for null or empty value
                if (value == null || string.IsNullOrEmpty(value.ToString()))
                {
                    return true;
                }
                else
                {
                    return false;
                }
            }
            else
            {
                // Parameter provided, convert parameter to boolean for comparison
                try
                {
                    bool comparing = System.Convert.ToBoolean(parameter);

                    if (comparing)
                    {
                        // Compare value with true condition
                        if (value == null || string.IsNullOrEmpty(value.ToString()))
                        {
                            return true;
                        }
                        else
                        {
                            return false;
                        }
                    }
                    else
                    {
                        // Compare value with false condition
                        if (value != null)
                        {
                            return true;
                        }
                        else
                        {
                            return false;
                        }
                    }
                }
                catch (Exception ex)
                {
                    throw new ArgumentException("Unable to convert value", ex);
                }
            }
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            return null;
        }
    }
}